### PR TITLE
Let's not change g-code while we're sending it...

### PR DESCRIPTION
**Pull request by thormj**
Short patch... don't let the user reload/load g-code while we're sending it.
Machine goes quite nuts.

### CODE REVIEW

**Comment by thormj** [comment]
And this indicates a further problem, but I'm not sure how to explain it:
If you're running "Corner-1.gc"
And you load in "Dragon.gc"
The red crosshairs stay on the path of Corner-1 until the cut is done.
The black crosshairs "follow" Dragon.
When the cut is done, the machine starts slewing towards where Dragon is, but the cursor on Dragon doesn't stop... it keeps moving.  If you press pause, the black crosshairs stop and the machine slewing stops,... and there's no way to tell the machine to "align yourself" except if the "slewing" is faster than the "cutting" on Dragon1.  And by slewing, I mean all axes are moving at maximum speed to get there (even Z).

This is the "odd behavior" that I saw when changing to "Prior-Z" using the controls.

File: groundcontrol.kv [hunk 1]
```diff
@@ -280,9 +280,11 @@
     Button:
         text: 'Open File'
         on_press: root.openFile()
+		disabled: app.data.uploadFlag == 1
```
Comment (BarbourSmith) [inline groundcontrol.kv]: I didn't you could reference data like this...why are we not doing this everywhere?...beautiful!

**Comment by BarbourSmith** [comment]
My computer is installing updates so I am a little behind on merging these, but I will merge as soon as it is finished

File: groundcontrol.kv (hunk 1 above)
Comment (thormj) [reply groundcontrol.kv]: I'm beginning to like Kivy... but I still get confused about self.x vs root.x vs app.x vs x
I tried root.data, but "root" is the class, and at construct time, it doesn't have "data" (that gets done just a bit later.

And I'm not sure you get to format it.  Text: "0.2f"%root.data.ZPos explodes.

Comment (BarbourSmith) [reply groundcontrol.kv]: I can't tell you how much time I've spent struggling with things not having access to data at construct time. I love kivy, but that was one thing that really bugged me...looks like I just missed the option to call app.data which would have made it easy.

Does it work from within modules too, or just the .kv file?

**Comment by thormj** [comment]
Should I make the "further problem" into an issue?
I tried (no cutting, just running, powered Z) to duplicate it without changing the files on the fly and was unsuccessful... but there was this one time... that it bit me...
But I'm not sure how to get it to bite me again without changing the file on it...
A clunky way to solve it is "when paused gets activated" the Pause button splits into two:
1. Resume Cut (resume the move, but do not feed any more lines) - this should (?) let the red/black crosshairs align (maybe?  I'm not sure what's going on when they differ)...
2. Resume G-Code (resume the sending of g-code)

But... like I said, I don't know what's going on here...  Just when the red and black move apart, bad juju can happen.

**Comment by BarbourSmith** [comment]
I like the idea of making an issue as a place to talk about the issue further.

Gocde can cause unpredictable behavior when you essentially copy and past the second half of one file onto the first half of another file by opening a new file midway through a run.

The real issue is that lines of gcode rely on information sent previously so a line that might not cause any trouble when the machine is in one state can have unwelcome effects when it is in a different state.

I think what I am not understanding is how this PR doesn't fix the issue. If you can't open a new file while one is running, how can the problem happen?

**Comment by thormj** [comment]
I found a way to reproduce it... moving the text for that to a new issue.

File: groundcontrol.kv (hunk 1 above)
Comment (thormj) [reply groundcontrol.kv]: I've only done it in the .kv file so far... that was slicker than expected....

Comment (thormj) [reply groundcontrol.kv]: On a completely different thread, can you tell me how saving data is supposed to work?  I think Issue #591 is close, but the load/save just isn't.

**Pull request merged**